### PR TITLE
Integrate skygear sdk part 2

### PR DESCRIPTION
**Pull request by carmenlau**
connect  #132
Used sdk completely

Updated command:
- config
    - set-cluster-server
- app
    - deploy
    - create
    - view-user-config
    - update-user-config
    - scaffold

### CODE REVIEW

File: src/container/CLIContainer.ts [hunk 1]
```diff
@@ -1,9 +1,63 @@
...
+      throw new Error('unexpected type');
+  }
+}
+
+function encodeHook(hook: HookConfig): any {
+  return {
+    async: hook.async,
+    event: hook.event,
+    path: hook.path || '',
+    timeout: hook.timeout
```
Comment (louischan-oursky) [inline src/container/CLIContainer.ts]: `async` and `timeout` are no longer needed

File: src/container/types.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export interface App {
   id: string;
   name: string;
+  last_deployment_id: string;
```
Comment (louischan-oursky) [inline src/container/types.ts]: Should be `last_deployment_id?: string`

File: src/container/types.ts [hunk 3]
```diff
@@ -32,3 +33,73 @@ export interface ForgotPasswordConfiguration {
   subject?: string;
   reply_to?: string;
 }
+
+export interface HttpHandlerConfig {
+  type: 'http-handler';
+  path: string;
+  hook?: HookConfig;
```
Comment (louischan-oursky) [inline src/container/types.ts]: Can be removed

File: src/container/types.ts [hunk 4]
```diff
@@ -32,3 +33,73 @@ export interface ForgotPasswordConfiguration {
...
+  dockerfile?: string;
+  secrets: string[];
+  environment: { [name: string]: string };
+}
+
+export interface HookConfig {
+  event: string;
+  async: boolean;
+  timeout: number;
+  path: string;
```
Comment (louischan-oursky) [inline src/container/types.ts:58]: Can remove `async` and `timeout`

**Comment by carmenlau** [comment]
Updated, thanks!

**Pull request merged**